src/Accordion/Accordion.tsx

```diff
@@ -1,95 +1,88 @@
+    const [id, setId] = React.useState<string>(props.id);
+    const [className, setClassName] = React.useState<string>("seb accordion");
 
-    React.useEffect(() => constructIds(), [props.list]);
-    React.useEffect(() => setActive(props.defaultExpanded), [props.defaultExpanded]);
+    /** Sets custom id if the user din't pass any */
     React.useEffect(() => setId(props.id || randomId("accordion-")), [props.id]);
-    React.useEffect(() => setClassName(classnames(["seb", "accordion", { alternative: props.alternative }, props.className])), [props.className, props.alternative]);
-
-    /** Constructs identifiers for accordion items to be used enable accessibility features */
-    const constructIds: VoidFunction = React.useCallback(() => {
-        const idListToSet: Array<string> = [];
-        props.list?.map(() => idListToSet.push(randomId("accordion-item-")));
-        setIdList(idListToSet);
-    }, [props.list, setIdList]);
+    React.useEffect(() => setClassName(classnames(["seb", "accordion", { alternative }, { inverted }, props.className])), [props.className, alternative, inverted]);
+    React.useEffect(() => {
+        typeof props.defaultValue === "number" && setActive(props.defaultValue);
+    }, [props.defaultValue]);
 
     /**
      * Handles accordion item click event
      * @param {React.MouseEvent<HTMLButtonElement>} e MouseEvent
      */
-    const onToggle: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
+    const onToggleInner: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
         (e: React.MouseEvent<HTMLButtonElement>) => {
-            setActive(active === Number(e.currentTarget.value) ? null : Number(e.currentTarget.value));
+            const index: number = Number(e.currentTarget.dataset.indexNumber);
+            !isNaN(index) && setActive(active === index ? -1 : index);
         },
-        [active, setActive]
+        [active]
     );
 
+    if (isNaN(active)) {
+        /**
+         * `active` is not initialized with `props.defaultValue` (`Number(undefined) === NaN`).
+         * Should only run once. The active state will never be set to NaN.
+         *
+         * At least one child is needed to determine the default.
+         */
+        if (list?.length || React.Children.count(props.children)) {
+            // If the default value is set from the parent, it should override anything else
+            let defaultValue: number = -1;
+            // Trying to find an item set to active in the list prop first
+            defaultValue = list ? list.findIndex((item: AccordionItemProps) => item.defaultChecked) : -1;
+            if (defaultValue === -1) {
+                // If the active is not found in one of the list items maybe it's one of the children
+                defaultValue = React.Children.toArray(props.children).findIndex((Child: React.ReactElement<AccordionItemProps>) => Child.props?.defaultChecked);
+                if (list) {
```

Comment:
i dont think thi is necessery because if defaultValue === -1 it means there is no list since you just checked it in line 58

Reply:
correct -> `if (list)` will never be true here.

Reply:
It is necessary, the reason is if there is a list but the index was not found in the list, the `defaultValue` will be `-1` even though the list exists. If it does exists I need to offset the index (`defaultValue`) with the length of the list.
see example
```tsx
<Accordion list={[ { header: "first", children: "Some content" } ]}>
    <AccordionItem defaultChecked header="Second">Some content</AccordionItem>
    <AccordionItem header="Second">Some content</AccordionItem>
</Accordion>
```
In this example ☝️ .. The default value should be `1` which is the index of the checked child `0` + the length of the list which is `1`. If you imagine having 3 rendered accordion items, then `1` would be the second.

Reply:
We agree you need to check for the possibility of  list existing and the index not being found but if you check the way you structured your code the if (list) will never be true. SO you have to restructure the code a different way. Maybe check the list at the beginning.